File: routes/capcha.js
var png = require('image/png');
var capcha = require('./lib/image/capcha.js');


var r = capcha.makeCapcha()
var img = r.img;
var str = r.str;

session.set('__verify', str);

response.header().set('content-type', 'image/png')
var err = png.encode(response, img)
if (err) {
  throw err;
}

File: routes/capcha.go
package routes

import (
	"image/png"

	"github.com/zengming00/go-testShop/framework"
	"github.com/zengming00/go-testShop/lib/image"
)

func Capcha(ctx *framework.HandlerContext) {
	var r = image.MakeCapcha()

	ctx.SetSessionVal("__verify", r.Str)

	ctx.W.Header().Set("content-type", "image/png")
	err := png.Encode(ctx.W, r.Img)
	if err != nil {
		panic(err)
	}
}
